src/carbonio-files-ui-common/hooks/useCreateNewActions.tsx

```diff
@@ -0,0 +1,209 @@
+/*
+ * SPDX-FileCopyrightText: 2024 Zextras <https://www.zextras.com>
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+import React, { useCallback, useEffect, useMemo } from 'react';
+
+import { Text } from '@zextras/carbonio-design-system';
+import { useTranslation } from 'react-i18next';
+
+import { useCreateFolderMutation } from './graphql/mutations/useCreateFolderMutation';
+import { useCreateModal } from './modals/useCreateModal';
+import { useCreateDocsFile } from './useCreateDocsFile';
+import { useHealthInfo } from './useHealthInfo';
+import { ACTION_IDS, ACTION_TYPES } from '../../constants';
+import { useActiveNode } from '../../hooks/useActiveNode';
+import { NewAction, useCreateOptions } from '../../hooks/useCreateOptions';
+import { DOCS_EXTENSIONS, FILES_APP_ID } from '../constants';
+import { DocsType } from '../types/common';
+import { GetChildrenQuery } from '../types/graphql/types';
+import { getDocumentGenericType, getNewDocumentActionLabel, isFolder } from '../utils/utils';
+
+export const useCreateNewActions = (
+	currentFolderId: string,
+	currentFolder: GetChildrenQuery | undefined,
+	canCreateFolder: boolean,
+	canCreateFile: boolean
+): NewAction[] => {
+	const { setActiveNode } = useActiveNode();
+	const { setCreateOptions, removeCreateOptions } = useCreateOptions();
+	const [t] = useTranslation();
+
+	const { createFolder } = useCreateFolderMutation();
+
+	const createFolderCallback = useCallback(
+		(_parentId: string, newName: string) => {
+			if (currentFolder?.getNode && isFolder(currentFolder.getNode)) {
+				return createFolder(currentFolder.getNode, newName).then((result) => {
+					if (result.data) {
+						setActiveNode(result.data.createFolder.id);
+					}
+					return result;
+				});
+			}
+			return Promise.reject(new Error('cannot create folder on invalid node'));
+		},
+		[createFolder, currentFolder?.getNode, setActiveNode]
+	);
+
+	const { openCreateModal: openCreateFolderModal } = useCreateModal();
+
+	const createFolderAction = useCallback(
+		(event: React.SyntheticEvent | KeyboardEvent) => {
+			event?.stopPropagation();
```

Comment:
can event be undefined here?
```suggestion
			event.stopPropagation();
```